### PR TITLE
Added automatic language and theme selection

### PR DESCRIPTION
**Pull request by EntenKoeniq**
Based on the browser settings

### CODE REVIEW

File: app/Http/Middleware/LocalizationMiddleware.php [hunk 1]
```diff
@@ -14,6 +15,17 @@ public function handle(Request $request, Closure $next): Response
...
+            $language_exist = WebsiteLanguage::where('country_code', '=', $country_code)->first();
+            if ($language_exist !== null) {
+                App::setLocale($country_code);
+                Session::put('locale', $country_code);
+            } else {
+                $default_locale = env('APP_LOCALE', 'en');
+                App::setLocale($default_locale);
+                Session::put('locale', $default_locale);
+            }
         }
```
Comment (DennisObject) [inline app/Http/Middleware/LocalizationMiddleware.php]:
```suggestion
        if (Session::has('locale')) {
    App::setLocale(Session::get('locale'));
    
    return $next($request);
}

$countryCode = strtolower(isset($_SERVER["HTTP_CF_IPCOUNTRY"]) ? $_SERVER["HTTP_CF_IPCOUNTRY"] : substr($_SERVER['HTTP_ACCEPT_LANGUAGE'], 0, 2));
if (WebsiteLanguage::where('country_code', '=', $countryCode)->doesntExist()) {
    $defaultLocale = env('APP_LOCALE', 'en');
    App::setLocale($defaultLocale);
    Session::put('locale', $defaultLocale);
    
    return $next($request);
}

App::setLocale($countryCode);
Session::put('locale', $countryCode);
```

**Comment by DennisObject** [comment]
Awesome! Thank you so much for the contribution - I added a suggestion for a little refactor of the code, if you agree with my comment, you can simply press "commit suggestion" and the code suggested will be added :)

File: app/Http/Middleware/LocalizationMiddleware.php (hunk 1 above)
Comment (EntenKoeniq) [reply app/Http/Middleware/LocalizationMiddleware.php]: Wrong return

You should use `return $next($request);` instead of `return;`

Comment (DennisObject) [reply app/Http/Middleware/LocalizationMiddleware.php]: Very true! good catch

**Review by DennisObject — APPROVED** [review]

**Pull request merged**